### PR TITLE
#170 dig amendmendts

### PR DESCRIPTION
**Pull request by mgorny-morecode**
Use input and output constants | Change docker image from `komand/python-2-slim-plugin:2` to `komand/python-3-37-slim-plugin:3` to reduce plugin image size | Added "f" strings | Remove duplicate code

### Description

Describe the proposed changes:

  -

### Testing

Any testing information goes here.

## PR Requirements

Developers, verify you have completed the following items by checking them off:

### Style

Review the [style guide](https://komand.github.io/python/style.html)

- [x] For dependencies, pin [OS package](https://komand.github.io/python/style.html#dockerfile) and [Python package](https://komand.github.io/python/style.html#requirements-txt) versions
- [x] For security, set least privileged account with ``USER nobody`` in the ``Dockerfile`` when possible
- [x] For size, use the [slim SDK images](https://komand.github.io/python/sdk.html#sdk-versions) when possible: ``komand/python-3-37-slim-plugin`` and ``komand/python-3-37-plugin``
- [x] For error handling, use of [PluginException](https://komand.github.io/python/error_handling.html#plugin-exceptions) and [ConnectionTestException](https://komand.github.io/python/error_handling.html#connection-exceptions)
- [x] For logging, use [self.logger](https://komand.github.io/python/sdk.html#logging)
- [x] For docs, use [changelog style](https://komand.github.io/python/style.html#changelog)
- [x] For docs, validate markdown with ``make validate`` which calls ``mdl`` to lint ``help.md``

### Functional Checklist
- [x] Work fully completed
- [x] Functional
  - [x] Any new actions/triggers include JSON [test files](https://komand.github.io/python/style.html#tests) in the `tests/` directory created with `icon-plugin run -c sample $action > tests/$action.json`
  - [x] Tests should all pass unless it's a negative test. Negative tests have a naming convention of `tests/$action_bad.json`
  - [x] Unsuccessful tests should fail by raising an exception causing the plugin to die and an object should be returned on successful test
  - [x] Add functioning test results to PR, sanitize any output if necessary
    * Single action/trigger `icon-plugin run -T tests/example.json --debug --jq`
    * All actions/triggers shortcut `icon-plugin run -T all --debug --jq` (use PR format at end)
  - [x] Add functioning run results to PR, sanitize any output if necessary
    * Single action/trigger `icon-plugin run -R tests/example.json --debug --jq`
    * All actions/triggers shortcut `icon-plugin run -R all --debug --jq` (use PR format at end)

### Run

<details>

```
{
  "body": {
    "log": "rapid7/Dig:1.0.3. Step name: forward\nExecuting command /usr/bin/dig google.com A\n",
    "meta": {},
    "output": {
      "all_answers": [
        "172.217.16.14"
      ],
      "answer": "172.217.16.14",
      "fulloutput": "\n; \u003c\u003c\u003e\u003e DiG 9.12.4-P2 \u003c\u003c\u003e\u003e google.com A\n;; global options: +cmd\n;; Got answer:\n;; -\u003e\u003eHEADER\u003c\u003c- opcode: QUERY, status: NOERROR, id: 64681\n;; flags: qr rd ra; QUERY: 1, ANSWER: 1, AUTHORITY: 0, ADDITIONAL: 1\n\n;; OPT PSEUDOSECTION:\n; EDNS: version: 0, flags:; udp: 4096\n;; QUESTION SECTION:\n;google.com.\t\t\tIN\tA\n\n;; ANSWER SECTION:\ngoogle.com.\t\t208\tIN\tA\t172.217.16.14\n\n;; Query time: 13 msec\n;; SERVER: 10.0.2.3#53(10.0.2.3)\n;; WHEN: Sat Jan 25 15:21:29 UTC 2020\n;; MSG SIZE  rcvd: 55\n\n",
      "last_answer": "172.217.16.14",
      "nameserver": "10.0.2.3",
      "question": "google.com",
      "status": "NOERROR"
    },
    "status": "ok"
  },
  "type": "action_event",
  "version": "v1"
}

```

<summary>
docker run --rm -i rapid7/dig:1.0.3 --debug run < tests/forward.json
</summary>
</details>

<details>

```
{
  "body": {
    "log": "rapid7/Dig:1.0.3. Step name: forward\nExecuting command /usr/bin/dig asdfasdfasdfasdfasdfads.com A\n",
    "meta": {},
    "output": {
      "all_answers": [
        "Not found"
      ],
      "answer": "Not found",
      "fulloutput": "Resolution failed, nameserver 10.0.2.3 returned NXDOMAIN status",
      "last_answer": "Not found",
      "nameserver": "10.0.2.3",
      "question": "asdfasdfasdfasdfasdfads.com",
      "status": "NXDOMAIN"
    },
    "status": "ok"
  },
  "type": "action_event",
  "version": "v1"
}

```

<summary>
docker run --rm -i rapid7/dig:1.0.3 --debug run < tests/forward_bad.json
</summary>
</details>

<details>

```
{
  "body": {
    "log": "rapid7/Dig:1.0.3. Step name: forward\nExecuting command /usr/bin/dig www.komand.com CNAME\n",
    "meta": {},
    "output": {
      "all_answers": [
        "Not found"
      ],
      "answer": "Not found",
      "fulloutput": "\n; \u003c\u003c\u003e\u003e DiG 9.12.4-P2 \u003c\u003c\u003e\u003e www.komand.com CNAME\n;; global options: +cmd\n;; Got answer:\n;; -\u003e\u003eHEADER\u003c\u003c- opcode: QUERY, status: NOERROR, id: 51920\n;; flags: qr rd ra; QUERY: 1, ANSWER: 0, AUTHORITY: 1, ADDITIONAL: 1\n\n;; OPT PSEUDOSECTION:\n; EDNS: version: 0, flags:; udp: 4096\n;; QUESTION SECTION:\n;www.komand.com.\t\t\tIN\tCNAME\n\n;; AUTHORITY SECTION:\nkomand.com.\t\t600\tIN\tSOA\tns-1502.awsdns-59.org. awsdns-hostmaster.amazon.com. 1 7200 900 1209600 86400\n\n;; Query time: 39 msec\n;; SERVER: 10.0.2.3#53(10.0.2.3)\n;; WHEN: Sat Jan 25 15:21:34 UTC 2020\n;; MSG SIZE  rcvd: 125\n\n",
      "last_answer": "Not found",
      "nameserver": "10.0.2.3",
      "question": "www.komand.com",
      "status": "NOERROR"
    },
    "status": "ok"
  },
  "type": "action_event",
  "version": "v1"
}

```

<summary>
docker run --rm -i rapid7/dig:1.0.3 --debug run < tests/forward_cname.json
</summary>
</details>

<details>

```
{
  "body": {
    "log": "rapid7/Dig:1.0.3. Step name: reverse\nExecuting command /usr/bin/dig -x 8.8.8.8\n",
    "meta": {},
    "output": {
      "answer": "dns.google",
      "fulloutput": "\n; \u003c\u003c\u003e\u003e DiG 9.12.4-P2 \u003c\u003c\u003e\u003e -x 8.8.8.8\n;; global options: +cmd\n;; Got answer:\n;; -\u003e\u003eHEADER\u003c\u003c- opcode: QUERY, status: NOERROR, id: 59923\n;; flags: qr rd ra; QUERY: 1, ANSWER: 1, AUTHORITY: 0, ADDITIONAL: 1\n\n;; OPT PSEUDOSECTION:\n; EDNS: version: 0, flags:; udp: 512\n;; QUESTION SECTION:\n;8.8.8.8.in-addr.arpa.\t\tIN\tPTR\n\n;; ANSWER SECTION:\n8.8.8.8.in-addr.arpa.\t80352\tIN\tPTR\tdns.google.\n\n;; Query time: 12 msec\n;; SERVER: 10.0.2.3#53(10.0.2.3)\n;; WHEN: Sat Jan 25 15:21:36 UTC 2020\n;; MSG SIZE  rcvd: 73\n\n",
      "nameserver": "10.0.2.3",
      "question": "8.8.8.8",
      "status": "NOERROR"
    },
    "status": "ok"
  },
  "type": "action_event",
  "version": "v1"
}

```

<summary>
docker run --rm -i rapid7/dig:1.0.3 --debug run < tests/reverse.json
</summary>
</details>

<details>

```
{
  "body": {
    "log": "rapid7/Dig:1.0.3. Step name: reverse\nExecuting command /usr/bin/dig -x 1.1.1.1.1\n",
    "meta": {},
    "output": {
      "answer": "Not found",
      "fulloutput": "Resolution failed, nameserver 10.0.2.3 returned NXDOMAIN status",
      "nameserver": "10.0.2.3",
      "question": "1.1.1.1.1",
      "status": "NXDOMAIN"
    },
    "status": "ok"
  },
  "type": "action_event",
  "version": "v1"
}

```

<summary>
docker run --rm -i rapid7/dig:1.0.3 --debug run < tests/reverse_bad.json
</summary>
</details>

### Test

Autogenerate with:
<details>

```
{
  "body": {
    "log": "rapid7/Dig:1.0.3. Step name: forward\n",
    "meta": {},
    "output": null,
    "status": "ok"
  },
  "type": "action_event",
  "version": "v1"
}

```

<summary>
docker run --rm -i rapid7/dig:1.0.3 --debug test < tests/forward.json
</summary>
</details>

Autogenerate with:
<details>

```
{
  "body": {
    "log": "rapid7/Dig:1.0.3. Step name: forward\n",
    "meta": {},
    "output": null,
    "status": "ok"
  },
  "type": "action_event",
  "version": "v1"
}

```

<summary>
docker run --rm -i rapid7/dig:1.0.3 --debug test < tests/forward_bad.json
</summary>
</details>

Autogenerate with:
<details>

```
{
  "body": {
    "log": "rapid7/Dig:1.0.3. Step name: forward\n",
    "meta": {},
    "output": null,
    "status": "ok"
  },
  "type": "action_event",
  "version": "v1"
}

```

<summary>
docker run --rm -i rapid7/dig:1.0.3 --debug test < tests/forward_cname.json
</summary>
</details>

Autogenerate with:
<details>

```
{
  "body": {
    "log": "rapid7/Dig:1.0.3. Step name: reverse\n",
    "meta": {},
    "output": null,
    "status": "ok"
  },
  "type": "action_event",
  "version": "v1"
}

```

<summary>
docker run --rm -i rapid7/dig:1.0.3 --debug test < tests/reverse.json
</summary>
</details>

Autogenerate with:
<details>

```
{
  "body": {
    "log": "rapid7/Dig:1.0.3. Step name: reverse\n",
    "meta": {},
    "output": null,
    "status": "ok"
  },
  "type": "action_event",
  "version": "v1"
}

```

<summary>
docker run --rm -i rapid7/dig:1.0.3 --debug test < tests/reverse_bad.json
</summary>
</details>

### Validate

Autogenerate with:
<details>

```
[*] Use ``make menu`` for available targets
[*] Including available Makefiles: ../tools/Makefiles/Helpers.mk ../tools/Makefiles/Colors.mk
--
[*] Running validators
[*] Validating plugin at .

[*] Running Integration Validators...
[*] Executing validator HelpValidator
[*] Executing validator ChangelogValidator
[*] Executing validator RequiredKeysValidator
[*] Executing validator UseCaseValidator
[*] Executing validator SpecPropertiesValidator
[*] Executing validator SpecVersionValidator
[*] Executing validator FilesValidator
[*] Executing validator TagValidator
[*] Executing validator DescriptionValidator
[*] Executing validator TitleValidator
[*] Executing validator VendorValidator
[*] Executing validator DefaultValueValidator
[*] Executing validator IconValidator
[*] Executing validator RequiredValidator
[*] Executing validator VersionValidator
[*] Executing validator DockerfileParentValidator
[*] Executing validator LoggingValidator
[*] Executing validator ProfanityValidator
[*] Executing validator AcronymValidator
[*] Executing validator JSONValidator
[*] Executing validator OutputValidator
[*] Executing validator RegenerationValidator
[*] Executing validator HelpInputOutputValidator
[*] Executing validator SupportValidator
[*] Plugin successfully validated!

----
[*] Total time elapsed: 218.347ms

[*] Validating spec with js-yaml
[SUCCESS] Passes js-yaml spec check


[*] Validating markdown...
[SUCCESS] Passes markdown linting

[*] Validating python files for style...
[SUCCESS] Passes flake8 linting

[*] Validating python files for security vulnerabilities...
[main]  INFO    profile include tests: None
[main]  INFO    profile exclude tests: None
[main]  INFO    cli include tests: None
[main]  INFO    cli exclude tests: None
[main]  INFO    running on Python 3.7.6
Run started:2020-01-25 15:25:07.261876

Test results:
        No issues identified.

Code scanned:
        Total lines of code: 333
        Total lines skipped (#nosec): 0

Run metrics:
        Total issues (by severity):
                Undefined: 0.0
                Low: 0.0
                Medium: 0.0
                High: 0.0
        Total issues (by confidence):
                Undefined: 0.0
                Low: 0.0
                Medium: 0.0
                High: 0.0
Files skipped (0):
[SUCCESS] Passes bandit security checks

```

<summary>
make validate
</summary>
</details>

Autogenerate with:
<details>

```
[*] Validating plugin with all validators at .

[*] Running Integration Validators...
[*] Executing validator HelpValidator
[*] Executing validator ChangelogValidator
[*] Executing validator RequiredKeysValidator
[*] Executing validator UseCaseValidator
[*] Executing validator SpecPropertiesValidator
[*] Executing validator SpecVersionValidator
[*] Executing validator FilesValidator
[*] Executing validator TagValidator
[*] Executing validator DescriptionValidator
[*] Executing validator TitleValidator
[*] Executing validator VendorValidator
[*] Executing validator DefaultValueValidator
[*] Executing validator IconValidator
[*] Executing validator RequiredValidator
[*] Executing validator VersionValidator
[*] Executing validator DockerfileParentValidator
[*] Executing validator LoggingValidator
[*] Executing validator ProfanityValidator
[*] Executing validator AcronymValidator
[*] Executing validator JSONValidator
[*] Executing validator OutputValidator
[*] Executing validator RegenerationValidator
[*] Executing validator HelpInputOutputValidator
[*] Executing validator SupportValidator
[*] Executing validator ExceptionValidator
[*] Executing validator CredentialsValidator
[*] Executing validator PasswordValidator
[*] Executing validator PrintValidator
[*] Executing validator ConfidentialValidator
[*] Executing validator DockerValidator
[*] Executing validator URLValidator
WARNING: URLs found that return a 4xx code. Verify they are publicly accessible and if not, update with a working URL.
violation: help.md[134]: https://www.iana.org/assignments/dns-parameters/dns-parameters.xhtml
violation: help.md[180]: https://www.iana.org/assignments/dns-parameters/dns-parameters.xhtml
[*] Plugin successfully validated!

----
[*] Total time elapsed: 19009.352ms

```

<summary>
icon-validate --all .
</summary>
</details>

### CODE REVIEW

File: dig/komand_dig/connection/connection.py [hunk 1]
```diff
@@ -9,6 +9,3 @@ def __init__(self):
 
     def connect(self, params):
         pass
-
-    def test(self):
```
Comment (lmilby-r7) [inline dig/komand_dig/connection/connection.py]: Keep this. I know its returning nothing but its needed for the platform.
